services/brig/src/Brig/Run.hs

```diff
@@ -103,3 +124,51 @@ lookupRequestIdMiddleware :: (RequestId -> Wai.Application) -> Wai.Application
 lookupRequestIdMiddleware mkapp req cont = do
   let reqid = maybe def RequestId $ lookupRequestId req
   mkapp reqid req cont
+
+cleanExpiredPendingInvitations :: AppIO ()
+cleanExpiredPendingInvitations = do
+  safeForever "cleanExpiredPendingInvitations" $ do
+    today <- utctDay <$> (liftIO =<< view currentTime)
+    for_ [0 .. 9] $ \i ->
+      cleanUpDay (addDays (- i) today)
+    let d :: Int = 24 * 60 * 60
+    randomSecs <- liftIO (round <$> randomRIO @Double (0.5 * fromIntegral d, fromIntegral d))
+    threadDelay (randomSecs * 1_000_000)
+  where
+    cleanUpDay :: Day -> AppIO ()
+    cleanUpDay day =
+      forTrackedExpirations day $ \exps -> do
+        expiredEntries <-
+          catMaybes
+            <$> ( for exps $ \(PendingActivationExpiration uid expiresAt tid) -> do
+                    isExpired <- (expiresAt <=) <$> (liftIO =<< view currentTime)
+                    if isExpired
+                      then do
+                        isPendingInvitation <- (Just PendingInvitation ==) <$> lookupStatus uid
+                        invExpired <- isNothing <$> Data.lookupInvitation tid (coerce uid)
+                        when (isPendingInvitation && invExpired) $ do
+                          API.deleteUserNoVerify uid
+                        pure (Just uid)
+                      else pure Nothing
+                )
+        unless (null expiredEntries) $
```

Comment:
Hm... this would delete rows for a table inside a pagination of query to that table.
Probably not a good idea... I'll move it outside of the pagination

Reply:
Done